pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -719,3 +595,38 @@ func (exp *flowExporter) findFlowType(conn flowexporter.Connection) uint8 {
 		return 0
 	}
 }
+
+// ValidateAndUpdateStats validates and updates the flow record given the connection
+// key. Caller is expected to grab lock.
+func (exp *flowExporter) ValidateAndUpdateStats(pqItem *flowexporter.ItemToExpire) {
+	// Update the stats in flow record after it is sent successfully
+	pqItem.PrevPackets = pqItem.Conn.OriginalPackets
+	pqItem.PrevBytes = pqItem.Conn.OriginalBytes
+	pqItem.PrevReversePackets = pqItem.Conn.ReversePackets
+	pqItem.PrevReverseBytes = pqItem.Conn.ReverseBytes
```

Comment:
why are we ignoring the TCP state here?

Reply:
the original methods doesn't include TCP state here, I didn't do change on that. Currently TCP state is being updated in `addOrUpdateConn`. Is there any purpose to move it here?

Reply:
Oh, I thought we could maintain the previous TCP state here to be consistent with bytes/packets, the fields that indicate the activeness of flow. Did not realize it is not there before.
Felt referring to an old connection from pqItem in the existing connection was weird and not readable in `addOrUpdateConn`.

`existingConn.PriorityQueueItem.Conn.TCPState != conn.TCPState`

Reply:
I think maybe we can add a field `prevTCPState` to pqItem struct?

Reply:
sounds good.